oe_theme.theme

```diff
@@ -2572,28 +2572,38 @@ function oe_theme_preprocess_select(&$variables) {
  * Implements hook_page_attachments_alter().
  */
 function oe_theme_page_attachments_alter(array &$attachments) {
+  // If the theme is configured to not use the theme favicon, bail out.
+  if (!theme_get_setting('favicon.use_default')) {
+    return;
+  }
+
   // Use different favicon based on theme component library.
   $active_theme = \Drupal::theme()->getActiveTheme();
   if ($active_theme->getName() === 'oe_theme' || array_key_exists('oe_theme', $active_theme->getBaseThemeExtensions())) {
+    $oe_theme_path = \Drupal::service('extension.list.theme')->getPath('oe_theme');
+    $active_theme_path = $active_theme->getPath();
     $component_library = theme_get_setting('component_library') ?? 'ec';
+    $favicon = base_path() . $active_theme_path . "/images/favicons/$component_library/favicon.ico";
     foreach ($attachments['#attached']['html_head_link'] as &$link) {
       if ($link[0]['rel'] !== 'icon') {
         continue;
       }
-      $link[0]['href'] = base_path() . $active_theme->getPath() . "/images/favicons/$component_library/favicon.ico";
+      $link[0]['href'] = file_exists($favicon) ? $favicon : base_path() . $oe_theme_path . "/images/favicons/$component_library/favicon.ico";
```

Comment:
`file_exists($favicon)` will always be FALSE because `base_path()` returns the path as seen by the web server.

I noticed it when I added a custom favicon.ico to the `ewcms_theme`
In this case, `file_exists` checks for `/build/profiles/custom/ewcms/themes/ewcms_theme/images/favicons/ec/favicon.ico`, and this path does not exist in the filesystem